src/ferc_xbrl_extractor/xbrl.py

```diff
@@ -84,9 +84,11 @@ def _dedupe_newer_report_wins(df: pd.DataFrame, primary_key: list[str]) -> pd.Da
     old_index = df.index.names
     return (
         df.reset_index()
+        .sort_values("report_date")
```

Comment:
We can `sort_values()` before doing the groupby because according to the [groupby docs](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.groupby.html) "Groupby preserves the order of rows within each group."

Reply:
One potentially non-deterministic outcome here is that if there are duplicate values of `report_date` within the column then which set of data ends up being "last" might not always be the same. Do we think there can ever be multiple filings within a group that have the same date?

Reply:
Yes! 

For some reason, I made this issue in the PUDL repo instead: https://github.com/catalyst-cooperative/pudl/issues/2822 but the short of it is "we need to use metadata from `rssfeed` to sort them better."